app/models/ability.rb

```diff
@@ -0,0 +1,12 @@
+class Ability
+  include CanCan::Ability
+  def initialize(user)
+    if user.role == 'admin'
+      can %i[read create update destroy], Boat, user_id: user.id
+      can %i[read create update destroy], Reservation, user_id: user.id
+    else
+      can :read, Boat, user_id: user.id
+      can %i[read create update], Reservation, user_id: user.id
+    end
+  end
+end
```

Comment:
- [ ] I created an admin user through the console, and that's how I was able to review the "add" and "delete" functionality.  

![image](https://github.com/CakeInTech/SailingSpot/assets/57421823/01a40f1e-3329-4b57-beb5-4bd918584979)


However, upon logging out and logging in again, I lost all my privileges for some reason:

![image](https://github.com/CakeInTech/SailingSpot/assets/57421823/0a523430-b02b-4623-9a5d-c62b7ef3d9b3)


Kindly take a look if this bug is showing up on your end. 👍

Reply:
It seems like when you log out and login again. Changes are not happening and when I reload then I get all the privileges, so to deal with this we added window.location.reload when we log in. Let me know if you have better suggestion. 😄